src/parse-cookies.ts

```diff
@@ -1,13 +1,39 @@
 const parseCookies = (s: string): { [key: string]: string } => {
   if (s.length === 0) return {};
   const parsed: { [key: string]: string } = {};
-  const pattern = new RegExp("\\s*;\\s*");
-  s.split(pattern).forEach((i) => {
-    const [encodedKey, encodedValue] = i.split("=");
-    const key = decodeURIComponent(encodedKey);
-    const value = decodeURIComponent(encodedValue);
-    parsed[key] = value;
-  });
+  const pairs = s.split(';')
+
+  for (let i = 0; i < pairs.length; i++) {
+    const pair = pairs[i];
+    const index = pair.indexOf('=')
+
+    // skip things that don't look like key=value
+    if (index < 0) {
+      continue;
+    }
+
+    const key = pair.substring(0, index).trim()
+
+    // only assign once
+    if (undefined == parsed[key]) {
+      let val = pair.substring(index + 1, pair.length).trim()
+
+      // quoted values
+      if (val[0] === '"') {
```

Comment:
doesn't look like the previous implementation does this. What use case is this for?